projects/cds/src/cms-components/profile-tag/profile-tag.injector.ts

```diff
@@ -0,0 +1,127 @@
+    this.startTracking$ = merge(pageLoaded$, consentChanged$);
+  }
+
+  injectScript(): Observable<Boolean[] | NgRouterEvent> {
+    this.addScript();
+    return this.addTracker().pipe(
+      switchMap(() => {
+        return this.isProfileTagLoaded$.pipe(
+          filter(Boolean),
+          switchMap(() => {
+            return this.startTracking$;
+          })
+        );
+      })
+    );
+  }
+
+  private pageLoaded(): Observable<NgRouterEvent> {
+    return this.router.events.pipe(
+      filter(event => event instanceof NavigationEnd),
+      tap(() => {
+        this.w.Y_TRACKING.push({ event: 'Navigated' });
+      }),
+      tap(() => 'called ytracking Navigated')
+    );
+  }
+
+  private consentChanged(): Observable<Boolean[]> {
+    return this.anonymousConsentsService.getConsents().pipe(
+      map((anonymousConsents: AnonymousConsent[]) => {
+        return anonymousConsents.map(anonymousConsent => {
+          return this.anonymousConsentsService.isConsentGiven(anonymousConsent);
+        });
+      }),
+      filter((consentsGranted: Boolean[]) => {
+        // TODO: For now we dont trigger a consent withdrawn as we do not do
+        // granular consents, and one consent withdrawn would lead all tracking to stop
+        // our system will anyway deny a request if an individual consent has been withdrawn
+        for (const consentGranted of consentsGranted) {
+          if (!consentGranted) {
+            return false;
+          }
+        }
+        return true;
+      }),
+      take(1),
+      tap(() => {
+        this.w.Y_TRACKING.push({ event: 'ConsentChanged', granted: true });
+      })
+    );
+  }
+
+  private addTracker(): Observable<string> {
+    return this.baseSiteService.getActive().pipe(
+      filter(Boolean),
+      tap((site: string) => {
+        const newConfig: ProfileTagJsConfig = {
+          ...this.config.cds.profileTag,
+          tenant: this.config.cds.tenant,
+          siteId: site,
+          spa: true,
+          profileTagEventReciever: this.profileTagEventTriggered.bind(this),
+        };
+        this.track(newConfig);
+      })
+    );
+  }
+
+  private addScript(): void {
+    const doc: Document = this.winRef.document;
+    const profileTagScript: HTMLScriptElement = doc.createElement('script');
+    profileTagScript.type = 'text/javascript';
+    profileTagScript.async = true;
+    profileTagScript.src = this.config.cds.profileTag.javascriptUrl;
+    doc.getElementsByTagName('head')[0].appendChild(profileTagScript);
+  }
+
+  private track(options: ProfileTagJsConfig) {
+    const q = this.w.Y_TRACKING.q || [];
+    q.push([options]);
+    this.w.Y_TRACKING.q = q;
+  }
+
+  private profileTagEventTriggered(profileTagEvent) {
+    switch (profileTagEvent.eventName) {
+      case 'Loaded':
+        this.isProfileTagLoaded$.next(true);
+        break;
+      default:
+        //add logger from spartacus. this.logger.info(`Unsupported Event ${profileTagEvent.eventName}`)
```

Comment:
In the future we will likely add more event types, so I guess I wouldn't want the customer to see this, because it is probably fine if there are unknown event types that an older version doesn't know about. So I would say this is only needed for debug.

Reply:
Ah ok. In that case you can do something like this:
```
import {isDevMode} from '@angular/core'
if (isDevMode()) {
console.log(...)
}
```